main.cpp

```diff
@@ -18,13 +18,19 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& _rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        if (const CStyleCastExpr *Expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+            	_rewriter.InsertText(Expr->getLParenLoc(), "static_cast");
+            	_rewriter.ReplaceText(Expr->getLParenLoc(), 1, "<");
+            	_rewriter.ReplaceText(Expr->getRParenLoc(), 1, ">(");

```

Comment:
Please describe the difference between the use of `getRParenLoc()` and `getSubExprAsWritten()->getBeginLoc()`.

Reply:
If we have this code:
```
int main() {
    float abc;
    int i = (int)abc;
    return 0;
}
```
and use `getSubExprAsWritten()->getBeginLoc()`, then we will get:
```
int main() {
    float abc;
    int i = static_cast<int)>(bc);
    return 0;
}
```
`getSubExprAsWritten()->getBeginLoc()` - Retrieve the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes introduced by semantic analysis. This function returns the position after "(int)" and it's "a". If we will use `getRParenLoc()`, it returns position after "int" and it's ")":
```
int main() {
    float abc;
    int i = static_cast<int>(abc);
    return 0;
}
```

Reply:
"This function returns the position after "(int)" and it's "a". If we will use getRParenLoc(), it returns position after "int" and it's ")""
Quite incorrect description.
- `getSubExprAsWritten()->getBeginLoc()` returns the starting position of the given subexpression, as it was written in the source, i.e. starting position "abc" - "a".
- `getRParenLoc()` returns the position of the closing parenthesis.